pkg/auth/providers/oidc/oidc_provider.go

```diff
@@ -515,3 +523,26 @@ func (o *OpenIDCProvider) getOIDCProvider(ctx context.Context, oidcConfig *v32.O
 	// This will perform discovery in the oidc library
 	return oidc.NewProvider(ctx, oidcConfig.Issuer)
 }
+
+func validateACR(oauth2Token *oauth2.Token, config *v32.OIDCConfig) bool {
+	// if we have no ACR configured, all values are accepted
+	if config.AcrValue == "" {
+		return true
+	}
+	tokenClaims := make(jwt.MapClaims)
+	_, _, err := new(jwt.Parser).ParseUnverified(oauth2Token.AccessToken, &tokenClaims)
+	if err != nil {
+		logrus.Infof("unable to parse acr from access token: %v", err)
+		return false
+	}
+	acr, ok := tokenClaims["acr"].(string)
+	if !ok {
+		logrus.Errorf("acr value could not be parsed")
+		return false
+	}
+	if acr != config.AcrValue {
+		logrus.Infof("acr value in token does not match configured acr value")
+		return false
+	}
+	return true
+}
```

Comment:
When returning a bool I usually prefer using the `IsXXX` convention, like `IsValidACR`. Not sure if in this case is still nice.

And it looks like we are not using anything else from the config, so we can probably pass directly the AcrValue:

```go
func IsValidACR(oauth2Token *oauth2.Token, acr string) bool {
    // ...
}
```

Btw looking at the code I think we can totally avoid this part.

Reply:
makes sense.  I've done some refactoring.